docs/udf.rst

```diff
@@ -0,0 +1,179 @@
+field ``description`` that is required. You can use this to better describe what a particular version
+of the UDF does, for example:
+
+.. code:: java
+
+    @Udf(description = "Returns a string that is a substring of this string. The"
+        + " substring begins with the character at the specified startIndex and"
+        + " extends to the end of this string.")
+    public String substring(final String value, final int startIndex)
+
+    @Udf(description = "Returns a string that is a substring of this string. The"
+        + " substring begins with the character at the specified startIndex and"
+        + " extends to the character at endIndex -1.")
+    public String substring(final String value, final int startIndex, final int endIndex)
+
+
+Example UDF class
+-----------------
+
+The class below creates a UDF named ``multiply``. As can be seen this UDF can be invoked in 3
+different ways:
+
+- with two int parameters returning a long result.
+- with two long parameters returning a long result.
+- with two double parameters returning a double result.
+
+.. code:: java
+
+    @UdfDescription(name = "multiply", description = "multiplies 2 numbers")
+    public class Multiply {
+
+      @Udf(description = "multiply two ints")
+      public long multiply(final int v1, final int v2) {
+        return v1 * v2;
+      }
+
+      @Udf(description = "multiply two longs")
+      public long multiply(final long v1, final long v2) {
+        return v1 * v2;
+      }
+
+      @Udf(description = "multiply two doubles")
+      public double multiply(final double v1, double v2) {
+        return v1 * v2;
+      }
+
+    }
+
+Supported Types
+---------------
+
+The types supported by UDFs are currently limited to:
+
+- int, Integer
+- boolean, Boolean
+- long, Long
+- double, Double
+- String
+- java.util.Map (must be a map of supported types)
+- java.util.List (must be a list of supported types)
+
+Note: Complex types are not currently supported
+
+
+=========
+Deploying
+=========
+
+To deploy your UDFs you need to create a jar containing all of the classes required by the UDFs.
+If you depend on third-party libraries then this should be an uber-jar containing those libraries.
+Once the jar is created you need to deploy it to each KSQL server instance. The jar should be copied
+to the ``ext`` directory that is part of the KSQL distribution.
+
+The jars in the ``ext`` directory are only scanned at start-up, so you will need to restart your
+KSQL server instances.
+
+
+=====
+Usage
+=====
+
+Once your UDFs are deployed you can call them in the same way you would invoke any of the KSQL
+builtin functions, for example, using the ``multiply`` example above:
+
+.. code:: sql
+
+    CREATE STREAM number_stream (int1 INT, int2 INT, long1 BIGINT, long2 BIGINT)
+      WITH (VALUE_FORMAT = 'JSON', KAFKA_TOPIC = 'numbers');
+
+    SELECT multiply(int1, int2), multiply(long1, long2) FROM number_stream;
```

Comment:
I presume that `multiply` is not case sensitive. If so, maybe have one of the invocations be `MULTIPLY`?